### ISSUE TITLE
YOLOv11: OBB Predictions not capturing entire length of long objects

### ISSUE THREAD

**Issue by fyan-ralcone** [ultralytics/ultralytics#20443]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I have been trying to train an OBB model (yolo11s-obb) to capture rows of blueberry bushes. No matter what I try during training, the predictions from the model never capture the full length of the rows. I've tried upping the value of the box hyperparameter during training but that didn't seem to help. Is there something I'm missing or are there restrictions for OBB predictions?

Below are two examples. The image with the blue boxes is from the model I trained, the other image shows the original annotations I made for that image.

![Image](https://github.com/user-attachments/assets/e45c63fa-32db-4edf-b9f9-713403079250)

![Image](https://github.com/user-attachments/assets/bb3e6a88-7edf-4306-94a1-b2e8f4dc6188)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @fyan-ralcone, thank you for reaching out and providing detailed information and example images! 🚀 This is an automated response to help you get the fastest possible support. An Ultralytics engineer will review your issue and assist you further soon.

We recommend reviewing the [Docs](https://docs.ultralytics.com/) for helpful [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions and best practices for OBB (Oriented Bounding Box) tasks are covered there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) if possible. This will help us understand and debug your situation efficiently.

If you're seeking advice on custom training, please include as much context as possible (dataset structure, sample images, and logs) and check out our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our community for more support and idea sharing:
- [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧
- [Discourse](https://community.ultralytics.com/) for in-depth Q&A
- [Subreddit](https://reddit.com/r/Ultralytics) to connect with other users

## Upgrade

Please ensure you are using the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in these up-to-date verified environments (with all dependencies, including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for helping us improve Ultralytics! 🥇

**Comment by Y-T-G** [comment]
If your objects are too big, you should use smaller `imgsz`.

You can also try using P6 model.

https://www.reddit.com/r/Ultralytics/comments/1gbyc47/detecting_objects_that_are_extra_small_or_extra/

https://github.com/Y-T-G/community/blob/main/cfg/obb/yolo11-p6.yaml

**Comment by fyan-ralcone** [comment]
literally thank you so much

**Comment by glenn-jocher** [comment]
You're welcome @fyan-ralcone! For detecting long linear features like blueberry rows, the P6 model suggestion is particularly helpful as it uses additional detection layers designed for larger scale ranges. With OBB detection, there can be challenges with very elongated objects that extend beyond typical detection windows. The image size adjustment (smaller `imgsz`) helps the model "see" more of the object within each processing window. You might also consider adjusting your training dataset annotations to ensure consistent labeling of these long features. If you continue having issues, exploring the anchor configuration or adjusting the image tiling approach during inference could provide further improvements.

**Comment by xiaoyao123456789** [comment]
@fyan-ralcone 您好，换成P6就直接解决了吗，我发现并没有P6预训练模型。我使用的P6yaml+yolov8-obb.pt尝试训练，结果仍然是框不全，尤其是这种长目标。

![Image](https://github.com/user-attachments/assets/f8e248da-1393-42fd-8bd7-d6c57c9da3ac)

**Comment by cqjnovo** [comment]
> [@fyan-ralcone](https://github.com/fyan-ralcone) 您好，换成P6就直接解决了吗，我发现并没有P6预训练模型。我使用的P6yaml+yolov8-obb.pt尝试训练，结果仍然是框不全，尤其是这种长目标。
> 
> ![Image](https://github.com/user-attachments/assets/f8e248da-1393-42fd-8bd7-d6c57c9da3ac)

请问你这个问题解决了吗，我最近也遇到了这种长目标检测不全的问题

**Comment by glenn-jocher** [comment]
Thanks for the follow-up—since this issue is closed, please open a new issue for visibility and include your exact train/predict commands, a few sample images with labels, your `data.yaml`, and the output of `yolo checks` (text, not screenshots).

Quick tips for long, slender OBB targets:
- Use YOLO11 OBB weights (not v8) and preserve geometry during training: `rect=True`, `mosaic=0`, and the OBB-recommended `imgsz=1024`.
- Ensure labels follow the YOLO OBB four-point format and note OBB angles are constrained to 0–90 degrees; after fixing labels, delete any `labels/*.cache`.
- For very large scenes, consider offline tiling/splitting before training/inference.

Example:
```bash
yolo obb train model=yolo11s-obb.pt data=your.yaml imgsz=1024 rect=True mosaic=0
```

See the OBB task guide for angle constraints and result formats in the [OBB docs](https://docs.ultralytics.com/tasks/obb/), dataset format notes in the [OBB datasets overview](https://docs.ultralytics.com/datasets/obb/), and DOTA-style image splitting in the [DOTA OBB dataset page](https://docs.ultralytics.com/datasets/obb/dota-v2/).
